TOOLS/lua/autocrop.lua

```diff
@@ -25,74 +25,114 @@
 --
 -- to mpv's arguments. This may be desirable to allow cropdetect more
 -- time to collect data.
+
 require "mp.msg"
+require 'mp.options'
 
 script_name = mp.get_script_name()
 cropdetect_label = string.format("%s-cropdetect", script_name)
 crop_label = string.format("%s-crop", script_name)
 
--- number of seconds to gather cropdetect data
-detect_seconds = tonumber(mp.get_opt(string.format("%s.detect_seconds", script_name)))
-if not detect_seconds then
+-- auto: bool - Whether to automatically execute the script.
+-- auto_delay: number - Delay of automatic tasks, avoid certain videos starting with a black screen.
+-- auto_duration: number - Only videos longer than this length (seconds) will perform automatic tasks.
+-- auto_percent: number[1-100] - Only videos with a playback progress later than this will perform automatic tasks.
+-- detect_seconds: number - How long (seconds) to gather cropdetect data.
+-- detect_min_ratio: number[0.0-1.0] - The ratio of the minimum acceptable clip size to the original.
+-- detect_limits: number[0-255] - Black threshold for cropdetect.
+local options = {
+    auto = "yes",
+    auto_delay = 4,
+    auto_duration = 60,
+    auto_percent = 1,
+    detect_limits = 24,
+    detect_min_ratio = 0.8,
     detect_seconds = 1
-end
+}
+read_options(options)
 
-function del_filter_if_present(label)
-    -- necessary because mp.command('vf del @label:filter') raises an
-    -- error if the filter doesn't exist
+function is_filter_exist(label)
     local vfs = mp.get_property_native("vf")
     for i,vf in pairs(vfs) do
         if vf["label"] == label then
-            table.remove(vfs, i)
-            mp.set_property_native("vf", vfs)
             return true
         end
     end
     return false
 end
 
-function autocrop_start()
+function remove_filter(label)
+    if is_filter_exist(label) then
+        mp.command(string.format('vf remove @%s', label))
+        return true
+    end
+    return false
+end
+
+function start_crop()
     -- exit if cropdetection is already in progress
     if timer then
         mp.msg.warn("already cropdetecting!")
         return
     end
 
     -- if there's a crop filter, remove it and exit
-    if del_filter_if_present(crop_label) then
+    if remove_filter(crop_label) then
         return
     end
 
+    detect_crop()
+end
+
+function detect_crop()
     -- insert the cropdetect filter
-    ret=mp.command(
+    ret = mp.command(
         string.format(
-            'vf add @%s:cropdetect=limit=%f:round=2:reset=0',
-            cropdetect_label, 24/255
+            'vf pre @%s:cropdetect=limit=%f:round=2:reset=0',
+            cropdetect_label, options.detect_limits/255
```

Comment:
sorry, didn't see that you had made this an option. I still think you should pass it as is to an %s format specifier and let the user decide what style to use.

Reply:
Very strange, the results are different.
options.detect_limit = 24

```
string.format(
            'vf pre @%s:cropdetect=limit=%s:round=2:reset=0',
            cropdetect_label, options.detect_limit
        )
```
![image](https://user-images.githubusercontent.com/50797982/82863989-1960ce80-9f56-11ea-80d2-782bea2b6fb0.png)

```
string.format(
            'vf pre @%s:cropdetect=limit=%f:round=2:reset=0',
            cropdetect_label, options.detect_limit/255
        )
```
![image](https://user-images.githubusercontent.com/50797982/82864066-35647000-9f56-11ea-8d83-c3163617d645.png)

Reply:
Ok, it looks like it has to do with the fact that your content is 10-bit. `vf_cropdetect.c` has code like

```
    if (s->limit < 1.0)
        s->limit *= (1 << desc->comp[0].depth) - 1;
```

so it maps [0,1) ->  [0,1023) for 10-bit and [0,1) -> [0,255) for 8-bit. [0,1) is relative where > 1 is absolute and depends on the bit depth.

In short, I guess the 24 / 255 that you have is in fact a good default, which is actually the same as ffmpeg's default.